### PR TITLE
Add `reset` unit for `combineEvents` method

### PR DESCRIPTION
**Pull request by doasync**

### CODE REVIEW

**Comment by doasync** [comment]
What about `repeat` or `once`? Do we need config for the second param?

File: combine-events/combine-events.test.js [hunk 1]
```diff
@@ -237,3 +246,315 @@ test('example from readme', () => {
...
+test('reset', () => {
+  const fn = jest.fn();
+
+  const event1 = createEvent();
+  const event2 = createEvent();
+  const event3 = createEvent();
+  const event4 = createEvent();
+  const event5 = createEvent();
+
+  const event = combineEvents([event1, event2, event3, event4, event5]);
```
Comment (sergeysova) [inline combine-events/combine-events.test.js:259]: No reset here

File: combine-events/combine-events.test.js [hunk 2]
```diff
@@ -237,3 +246,315 @@ test('example from readme', () => {
     ]
   `);
 });
+
+test('reset', () => {
```
Comment (sergeysova) [inline combine-events/combine-events.test.js:250]: Test with the same name already exists

File: combine-events/index.d.ts [hunk 3]
```diff
@@ -10,3 +10,15 @@ export function combineEvents<Shape>(
 ): Event<
   { [K in keyof Shape]: Shape[K] extends Event<infer U> ? U : Shape[K] }
 >;
+export function combineEvents<Shape extends Tuple>(
+  shape: Shape,
+  reset: Unit<any>,
```
Comment (sergeysova) [inline combine-events/index.d.ts:15]: I think it is better to add configuration.

`{ reset }`

**Comment by sergeysova** [comment]
`repeat` and `once` could be added later

File: combine-events/index.d.ts (hunk 3 above)
Comment (sergeysova) [reply combine-events/index.d.ts:15]: And also the object form:

```ts
combineEvents({
  shape: {...},
  reset: unit,
})
```

Comment (doasync) [reply combine-events/index.d.ts:15]: How are you going to differentiate if the first object is `source` or `config`?

**Comment by sergeysova** [comment]
I think it can be refactored to object arguments form. Like this https://github.com/sergeysova/patronum/pull/43/files#diff-41f71b00da54afac96b38389da09d98eR4-R5

Ref #33

File: combine-events/index.d.ts (hunk 3 above)
Comment (doasync) [reply combine-events/index.d.ts:15]: I like the idea to use `combineEvents(config)`

```js
combineEvents({
  source: {...},
  reset: unit, 
  target,
})
```

No more signatures needed, config only.

**Comment by sergeysova** [comment]
Already implemented

**Pull request closed without merging**